### DISCUSSION TITLE
What happens to failed auto-retried jobs?

### DISCUSSION THREAD

**Discussion by olivierlacan** [OptimalBits/bull#2154] [Q&A] [answered]
[The job lifecycle diagram](https://optimalbits.github.io/bull/#a-jobs-lifecycle) in the docs is a bit confusing because it implies that failed auto-retried jobs become delayed jobs (which makes sense when thinking of a backoff strategy) but there's no Delayed callback to independently capture when a once failed job becomes a delayed job? 

<img width="521" alt="image" src="https://user-images.githubusercontent.com/65950/131925394-04e046c6-a6f8-435e-9fe0-13c7d9b5b0bf.png">

I'm trying to capture a queue depth metric and it's important for me to capture both jobs that are waiting for an idle worker and jobs that are waiting because they failed and therefore became delayed. 

The graph doesn't suggest that Delayed jobs ever trigger the Wait event but perhaps that's because the graph is mistaken? Or does a delayed job skip past the wait event altogether when its due time comes?

**Comment by manast** [comment, marked as answer]
Yes, thats right, currently delayed jobs are moved back to wait then their timer expire. So the diagram is wrong in that regard, however we have a new design in the works where the delayed jobs will be moved directly to active. Not sure it is worth to change the diagram now considering this.
